### ISSUE TITLE
Auto decline nixConfig values if not in interactive tty

### ISSUE THREAD

**Issue by jonringer** [NixOS/nix#7071]
**Describe the bug**
If you have a `nixConfig` value in a flake, then an unattended `nix` command will just poll forever for the prompt logic. It would be a good user experience for the warning to emit that `--accept-flake-config` can be passed to auto-accept rather than just quietly declining the values (currently it just hangs in a prompt).

**Steps To Reproduce**

1. Add `nixConfig`config value to a flake
2. Run a CI job running any `nix` command.

**Expected behavior**

```
warning: Auto-declining value "foo", please use `--accept-flake-config` to auto-accept values
```

Nix version: 2.10.3

**Additional context**

N/A

**Comment by bryanhonof** [comment]
I agree with this. It'd be good UX to tell them what they _could_ do. But _when_ do we tell them? If there's a way to detect that they're running in a non-interactive shell, I'd do it then. Otherwise just print the message before the first prompt?

**Comment by zimbatm** [comment]
Alternatively, if the proposed configuration is in the user configuration, it should be assumed that it's trusted. That way it would allow preparing the system/user configuration before using the flake and bypassing that issue.

**Comment by brainrake** [comment]
arguably, `accept-flake-config` should not be accepted from a flake, since that allows running commands with root without even a warning or prompt. `trusted-users` gives passwordless root so should probably be restricted to a few nix options.
https://github.com/NixOS/nix/issues/9649
